### PR TITLE
Fix/raptor rpc

### PR DESCRIPTION
**Pull request by andre-merzky**

### CODE REVIEW

File: src/radical/pilot/task_manager.py [hunk 1]
```diff
@@ -540,20 +540,20 @@ def add_pilots(self, pilots):
...
                     pilot_dict = pilot.as_dict()
-                    # real object: subscribe for state updates
                     pilot.register_callback(self._pilot_state_cb)
 
                 pid = pilot_dict['uid']
 
                 if pid in self._pilots:
                     raise ValueError('pilot %s already added' % pid)
-                self._pilots[pid] = pilot_dict
+                self._pilots[pid] = pilot
```
Comment (mtitov) [inline src/radical/pilot/task_manager.py:556]: should be kept `pilot_dict`

File: src/radical/pilot/pilot.py [hunk 2]
```diff
@@ -756,7 +756,7 @@ def _control_cb(self, topic, msg_data):
 
     # --------------------------------------------------------------------------
     #
-    def rpc(self, cmd, *args, **kwargs):
+    def rpc(self, cmd, *args, rpc_addr=None, **kwargs):
```
Comment (mtitov) [inline src/radical/pilot/pilot.py:759]: the ordering of inputs is not consistent with `Component`
btw, maybe we can not to have `*args`, that's brings some confusion, and instead to have `args=` explicitly

File: src/radical/pilot/agent/scheduler/continuous_ddmd.py
Comment (mtitov) [inline src/radical/pilot/agent/scheduler/continuous_ddmd.py:1]: should we exclude it from this PR?

File: src/radical/pilot/raptor_tasks.py [hunk 3]
```diff
@@ -97,16 +95,12 @@ def rpc(self, rpc: str,
...
+        kwargs['raptor_cmd'] = cmd
 
-        if not args:
-            args = dict()
-
-        args['raptor_cmd'] = rpc
-
-        return self._tmgr.pilot_rpc(self._pilot, cmd, args)
+        return self._tmgr.pilot_rpc(self._pilot, 'raptor_rpc', *args,
+                                    rpc_addr=self.uid, **kwargs)
```
Comment (mtitov) [inline src/radical/pilot/raptor_tasks.py:103]: in sync with comment above

**Review by mtitov — COMMENTED** [review]
overall looks good, but inputs for rpc-methods need to keep aligned

File: src/radical/pilot/task_manager.py (hunk 1 above)
Comment (mtitov) [reply src/radical/pilot/task_manager.py:556]: I was wrong here

File: src/radical/pilot/agent/scheduler/continuous_ddmd.py
Comment (andre-merzky) [reply src/radical/pilot/agent/scheduler/continuous_ddmd.py:1]: Yes, that was a mistake...  thanks

File: src/radical/pilot/pilot.py (hunk 2 above)
Comment (andre-merzky) [reply src/radical/pilot/pilot.py:759]: We discussed this out of band.  The current version allows a more native-like call invocation, thus leaving as is.

File: src/radical/pilot/raptor_tasks.py (hunk 3 above)
Comment (andre-merzky) [reply src/radical/pilot/raptor_tasks.py:104]: Thanks for spotting this, fixed.

**Comment by andre-merzky** [comment]
> overall looks good, but inputs for rpc-methods need to keep aligned

Ack, made this consistent now - well spotted!

File: src/radical/pilot/agent/scheduler/continuous_ddmd.py
Comment (mtitov) [reply src/radical/pilot/agent/scheduler/continuous_ddmd.py:1]: ping for cleanup

File: src/radical/pilot/task_manager.py [hunk 4]
```diff
@@ -660,7 +660,7 @@ def pilot_rpc(self, pid, cmd, args):
         if pid not in self._pilots:
             raise ValueError('tmgr does not know pilot %s' % pid)
 
-        return self._pilots[pid].rpc(cmd=cmd, args=args)
+        return self._pilots[pid].rpc(cmd=cmd, *args, rpc_addr=rpc_addr, **kwargs)
```
[the remark this answers is not in the document]
Comment (andre-merzky) [reply src/radical/pilot/task_manager.py:663]: right...

**Review by mtitov — APPROVED** [review]
LGTM!

**Pull request merged**